cmds/core/lsdrivers/lsdrivers_linux_test.go

```diff
@@ -0,0 +1,423 @@
+		{t: "d", n: filepath.Join(dir, "sys/bus/memory/devices/"), l: ""},
+		{t: "l", n: filepath.Join(dir, "sys/bus/memory/devices/memory3"), l: "../../../devices/system/memory/memory3"},
+		{t: "l", n: filepath.Join(dir, "sys/bus/memory/devices/memory12"), l: "../../../devices/system/memory/memory12"},
+		{t: "l", n: filepath.Join(dir, "sys/bus/memory/devices/memory1"), l: "../../../devices/system/memory/memory1"},
+		{t: "l", n: filepath.Join(dir, "sys/bus/memory/devices/memory10"), l: "../../../devices/system/memory/memory10"},
+		{t: "l", n: filepath.Join(dir, "sys/bus/memory/devices/memory8"), l: "../../../devices/system/memory/memory8"},
+		{t: "l", n: filepath.Join(dir, "sys/bus/memory/devices/memory6"), l: "../../../devices/system/memory/memory6"},
+		{t: "l", n: filepath.Join(dir, "sys/bus/memory/devices/memory15"), l: "../../../devices/system/memory/memory15"},
+		{t: "l", n: filepath.Join(dir, "sys/bus/memory/devices/memory4"), l: "../../../devices/system/memory/memory4"},
+		{t: "l", n: filepath.Join(dir, "sys/bus/memory/devices/memory13"), l: "../../../devices/system/memory/memory13"},
+		{t: "l", n: filepath.Join(dir, "sys/bus/memory/devices/memory2"), l: "../../../devices/system/memory/memory2"},
+		{t: "l", n: filepath.Join(dir, "sys/bus/memory/devices/memory11"), l: "../../../devices/system/memory/memory11"},
+		{t: "l", n: filepath.Join(dir, "sys/bus/memory/devices/memory0"), l: "../../../devices/system/memory/memory0"},
+		{t: "l", n: filepath.Join(dir, "sys/bus/memory/devices/memory9"), l: "../../../devices/system/memory/memory9"},
+		{t: "l", n: filepath.Join(dir, "sys/bus/memory/devices/memory7"), l: "../../../devices/system/memory/memory7"},
+		{t: "l", n: filepath.Join(dir, "sys/bus/memory/devices/memory5"), l: "../../../devices/system/memory/memory5"},
+		{t: "l", n: filepath.Join(dir, "sys/bus/memory/devices/memory14"), l: "../../../devices/system/memory/memory14"},
+		{t: "d", n: filepath.Join(dir, "sys/bus/memory/drivers/"), l: ""},
+	}
+	for _, f := range files {
+		switch f.t {
+		case "d":
+			if err := os.MkdirAll(f.n, 0755); err != nil {
+				t.Fatal(err)
+			}
+		case "l":
+			if err := os.Symlink(f.l, f.n); err != nil {
+				t.Fatal(err)
+			}
+		case "-":
+			f, err := os.Create(f.n)
+			if err != nil {
+				t.Fatal(err)
+			}
+			f.Close()
+		}
+	}
+
+	bus := filepath.Join(dir, "sys/bus")
+	out, err := lsdrivers(bus, false)
+	if err != nil {
+		t.Fatal(err)
+	}
+	o := strings.Join(out, "\n")
+	used := "acpi.button\npci.ahci\npci.ata_generic\npci.ata_piix\npci.mptspi\npci.nvme\npci.pata_sis\npci.sata_mv\npci.sata_nv\npci.sata_promise\npci.sata_sil\npci.sata_sis\npci.sata_svw\npci.sata_uli\npci.sata_via\npci.sata_vsc\npci.shpchp\npci.virtio-pci\npci.vmw_pvscsi\npci.vmxnet3\nplatform.alarmtimer\nplatform.pcspkr\nplatform.serial8250\npnp.rtc_cmos\npnp.serial\npnp.system\nscsi.sd\nscsi.sr\nusb.hub\nusb.usbfs\nusb.usbhid\nvirtio.virtio_net\nvirtio.virtio_rng\nvirtio.vmw_vsock_virtio_transport"
```

Comment:
Can you make this nicer to read?

Reply:
Also, can you split up the list above and have tests corresponding to the individual items here? I have no idea how to tell which corresponds to which.

Reply:
done